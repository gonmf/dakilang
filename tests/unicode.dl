a(["A\x42C", "A\u2713C"]).
a("A\x42C", "A\u2713C").
a("A\x42C", "A\u2713C") :- a(["A\x42C", "\u2713"]); a(["A\x42C", "\u2713"]).

a(X, Y) :- char(X, Y).
a(0x2713, Y)?

b(X, Y) :- char(X, Y).
b(0x42, Y)?

c(X, Y) :- ord(X, Y).
c('ðŸ˜‡', Y)?

print('Hello World \u{1F30E}', _)!
print('Hello World \u2764', _)!
print('Hello World \u2764\uFE0F', _)!
