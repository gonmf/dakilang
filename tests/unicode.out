> a(["A\x42C", "A\u2713C"]).
> a("A\x42C", "A\u2713C").
> a("A\x42C", "A\u2713C") :- a(["A\x42C", "\u2713"]); a(["A\x42C", "\u2713"]).
>
> a(X, Y) :- char(X, Y).
> a(0x2713, Y)?
Y = '‚úì'

>
> b(X, Y) :- char(X, Y).
> b(0x42, Y)?
Y = 'B'

>
> c(X, Y) :- ord(X, Y).
> c('üòá', Y)?
Y = 128519

>
> print('Hello World \u{1F30E}', _)!
Hello World üåé
Yes

> print('Hello World \u2764', _)!
Hello World ‚ù§
Yes

> print('Hello World \u2764\uFE0F', _)!
Hello World ‚ù§Ô∏è
Yes

